### PR TITLE
Added security context fixes

### PR DESCRIPTION
**Pull request by VenkataChalla554**
## Context
Added security context  fixes to welcome app , ingress , prometheus ,thanos and grafana

## Changes proposed in this pull request.

Added security_context  to container with following attributes.

`
  security_context {
            run_as_user  = 1000
            run_as_group = 3000
            capabilities {
              add  = ["NET_BIND_SERVICE"]
              drop = ["ALL"]
            }
            allow_privilege_escalation = false
            privileged                 = false
            run_as_non_root            = true
            read_only_root_filesystem  = true
            seccomp_profile {
              type = "RuntimeDefault"
            }
          }
`
 
## Guidance to review
Deployed on cluster2 , this can be tested.
https://www.cluster2.development.teacherservices.cloud/
https://prometheus.cluster2.development.teacherservices.cloud/
https://thanos.cluster2.development.teacherservices.cloud/
https://grafana.cluster2.development.teacherservices.cloud/
https://register-2222.cluster2.development.teacherservices.cloud/

## Before merging
Need to apply all environments before merge.

## After merging
Check all pipelines are success.

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by VenkataChalla554** [comment]
[Claims: worker heartbeat monitor](https://trello.com/c/PQgfB2yQ/1961-claims-worker-heartbeat-monitor)

**Comment by VenkataChalla554** [comment]
[Apply security fixes to apps deployed outside the terraform module](https://trello.com/c/j20bMVCA/1942-apply-security-fixes-to-apps-deployed-outside-the-terraform-module)

**Comment by saliceti** [comment]
I unerstand the NET_BIND_SERVICE capability is for binding to privileged ports. Can we listen to unprivileged ports so we can drop NET_BIND_SERVICE as well?

**Comment by VenkataChalla554** [comment]
> I unerstand the NET_BIND_SERVICE capability is for binding to privileged ports. Can we listen to unprivileged ports so we can drop NET_BIND_SERVICE as well?

Only thanos required NET_BIND_SERVICE  because it is using port 10901

File: cluster/terraform_kubernetes/welcome_app.tf [hunk 1]
```diff
@@ -114,3 +167,29 @@ resource "kubernetes_ingress_v1" "welcome_app" {
     }
   }
 }
+
+resource "kubernetes_config_map" "nginx_config" {
```
Comment (saliceti) [inline cluster/terraform_kubernetes/welcome_app.tf:171]: Can we rename to welcome_app_nginx_config ?

File: cluster/terraform_kubernetes/welcome_app.tf [hunk 2]
```diff
@@ -114,3 +167,29 @@ resource "kubernetes_ingress_v1" "welcome_app" {
...
+  metadata {
+    name      = "nginx-config"
```
Comment (saliceti) [inline cluster/terraform_kubernetes/welcome_app.tf:173]: Can we rename to welco-app-nginx-config ?

File: cluster/terraform_kubernetes/welcome_app.tf (hunk 1 above)
Comment (VenkataChalla554) [reply cluster/terraform_kubernetes/welcome_app.tf]: fixed

File: cluster/terraform_kubernetes/welcome_app.tf (hunk 2 above)
Comment (VenkataChalla554) [reply cluster/terraform_kubernetes/welcome_app.tf:173]: fixed

File: cluster/terraform_kubernetes/ingress_controller.tf [hunk 3]
```diff
@@ -157,30 +157,51 @@ resource "helm_release" "ingress-nginx" {
...
+    value = "1000"
+    type  = "auto"
+  }
+  set {
+    name  = "controller.podSecurityContext.runAsGroup"
+    value = "3000"
+    type  = "auto"
+  }
+  set {
+    name  = "controller.securityContext.capabilities.drop[0]"
```
[the remark this answers is not in the document]
Comment (VenkataChalla554) [reply cluster/terraform_kubernetes/ingress_controller.tf:172]: Added comment :   // By default, NET_BIND_SERVICE is added to the deployment by the Helm chart, even if we do not explicitly set it.

**Review by saliceti — APPROVED** [review]

**Pull request merged**